### PR TITLE
Add YAML tests for div and mod operations

### PR DESCRIPTION
**Pull request by dthaler**
Some of these illustrate current bugs, which is issue #382 

Signed-off-by: Dave Thaler <dthaler@microsoft.com>

### CODE REVIEW

File: test-data/divmod.yaml [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+
+code:
+  <start>: |
+    r1 %= r2  ; this could be modulo 0
+
+post:
+  - r1.type=number
+  - r2.type=number
+---
+test-case: negative modulo negative
```
Comment (elazarg) [inline test-data/divmod.yaml:269]:
```suggestion
test-case: negative modulo positive
```

File: test-data/divmod.yaml [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+
+pre: ["r1.type=number", "r1.value=-13"]
+
+code:
+  <start>: |
+    r1 %= 3
+
+post:
+  - r1.type=number
+  - r1.value=-1
```
Comment (elazarg) [inline test-data/divmod.yaml:279]: This (and the other results below) are not what they should be according to the specification draft, instead they work like `-(abs(a)%abs(b))`.

File: test-data/divmod.yaml (hunk 1 above)
Comment (elazarg) [reply test-data/divmod.yaml:269]: Missing test: positive modulo positive.

**Review by elazarg — COMMENTED** [review]
Looks good overall.

File: test-data/divmod.yaml (hunk 2 above)
Comment (dthaler) [reply test-data/divmod.yaml:279]: You're right... according to https://en.wikipedia.org/wiki/Modulo_operation the draft should say trunc() not floor().  I will fix that in the draft.

File: test-data/divmod.yaml (hunk 1 above)
Comment (dthaler) [reply test-data/divmod.yaml]: Done

File: test-data/divmod.yaml (hunk 2 above)
Comment (dthaler) [reply test-data/divmod.yaml:291]: Corrected draft

**Comment by elazarg** [comment]
@dthaler I have merged #384 by mistake. I can revert if that will be helpful.

**Comment by dthaler** [comment]
> @dthaler I have merged #384 by mistake. I can revert if that will be helpful.

No need, this one just tests the "broken" functionality, and 384 fixes the code and tests with "correct" brhavior, so I will close this PR.

**Pull request closed without merging**